stm/stm.ml

```diff
@@ -2432,11 +2432,7 @@ let observe ~doc id =
 let finish ~doc =
   let head = VCS.current_branch () in
   let () = observe ~doc (VCS.get_branch_pos head) in
-  let tip = VCS.cur_tip () in
-  if not (State.is_cached ~cache:true tip) then
-    CErrors.anomaly Pp.(str "Stm.finish: tip not cached");
```

Comment:
FYI, I tried to keep those three lines but the memory problem remained.

Reply:
Since it uses `cache:true` there can be a significant side effect

Reply:
The effect size is still surprising though

Reply:
I see the problem: `finish` is not only called by clients at the end of the document, but also at each `VtNow` command (eg `Notation`). So by forcing the cache there, we blow up memory.

I'll provide an alternative PR soon.